src/ansys/aedt/core/visualization/advanced/rcs_visualization.py

```diff
@@ -0,0 +1,1991 @@
+                norm_vect = [0, 0, 1]
+            actor = actor.ribbon(width=geo_width / 2, normal=norm_vect)
+        elif plot_type_lower == "projection":
+            if scene_actors is None:
+                return None
+            actor = pv.PolyData()
+            for model_actor in scene_actors.values():
+                mesh = model_actor.custom_object.get_mesh()
+                xypoints = mesh.points
+                xpos_ypos = np.column_stack((xpos, ypos, plot_data))
+                _, all_indices = self.__find_nearest_neighbors(xpos_ypos, xypoints)
+                mag_for_color = np.ndarray.flatten(cpos[all_indices])
+                if not mesh.__class__.__name__ == "PolyData":
+                    mesh_triangulated = mesh.triangulate()
+                    model_actor.custom_object.mesh = pv.PolyData(mesh_triangulated.points, mesh_triangulated.cells)
+                else:
+                    model_actor.custom_object.mesh.clear_data()
+                model_actor.custom_object.mesh[data_conversion_function] = mag_for_color
+                actor += model_actor.custom_object.mesh
+        return actor
+
+    @staticmethod
+    def stretch_data(data, scaling_factor, offset):
+        """
+        Stretches and scales the input data to a specified range.
+
+        This method normalizes the input data between its minimum and maximum values and then applies
+        a linear transformation using the formula: ``scaled_data = (data - min) / (max - min) * m + b``.
+        The parameters ``m`` and ``b`` control the scaling and shifting of the normalized data.
+
+        Parameters
+        ----------
+        data : numpy.ndarray or pandas.Series
+            The input data array or series to be stretched.
+        scaling_factor : float
+            The scaling factor applied to the normalized data.
+        offset : float
+            The offset added to the scaled data after normalization.
+
+        Returns
+        -------
+        numpy.ndarray or pandas.Series
+            Transformed data
+
+        Example:
+        -------
+        >>> data = np.array([1, 2, 3, 4, 5])
+        >>> stretched_data = stretch_data(data,2,1)
+        >>> print(stretched_data)
+        [1.  1.5 2.  2.5 3. ]
+        """
+        return (data - data.min()) / (data.max() - data.min()) * scaling_factor + offset
+
+    @staticmethod
+    def __find_nearest_neighbors(xpos_ypos, xypoints):
+        # Calculate squared Euclidean distance between each point in xypoints and xpos_ypos
+        distances = np.sqrt(((xpos_ypos[:, np.newaxis] - xypoints) ** 2).sum(axis=2))
+
+        # Find the index of the nearest neighbor for each query point in xypoints
+        all_indices = np.argmin(distances, axis=0)
+
+        # Get the distance for each query point (optional, can be removed if not needed)
+        nearest_distances = np.min(distances, axis=0)
+
+        return nearest_distances, all_indices
```

Comment:
Since `__find_nearest_neighbors` is used once and only `all_indices` is leveraged, I would refactor this method by:
1. removing `nearest_distances = np.min(distances, axis=0)`
2. removing the call to `np.sqrt`.
Since you are only interested in the nearest neighbors and not the distance value, performing the sqrt on every value isn't required (this will decrease the complexity of the method)